### ISSUE TITLE
Provide dockable Search window

### ISSUE THREAD

**Issue by fpunktk** [geany/geany#448]
I'd like to have a docked search panel instead of the search window that always covers the wrong part of my screen. 
The search panel could open at the bottom of the editor window on pressing ctrl+f and it can be closed when pressing esc (like it is now done with the search window).

**Comment by elextr** [comment]
I have re-defined ctrl-f to go to the toolbar search which works more like modern searches, ie finds progressively as you type, then find-next works immediately.

Or you could write a plugin to put a search tab in the message window.

**Comment by fpunktk** [comment]
thats a good workaround :-) the option that has to be changes is in the config dialog > shortcuts > focus (I have to guess the names because my ui is in german)

I'd still like to have a dockable search (and replace) panel.

**Comment by pik** [comment]
I agree - I too find the dock style find & replace to be a lot more convenient. i.e. https://www.dropbox.com/s/ovbnnc9tjfiutb8/2015-04-09-1428615448_1600x826.png

I'm not sure adding it to the message_window is the best choice, that might take up too much space and it would also require switching to it from other message_window tabs. Could a plugin add an additional tool-bar or would that behave poorly for some reason?

**Comment by elextr** [comment]
@pik I don't think plugins can add stuff to the toolbar :(  Stuff not already defined I mean.

**Comment by codebrainz** [comment]
I think the latex and scope plugins add stuff to the toolbar. IIRC they use GtkUIManager to merge their toolbar XML into Geany's or something.

**Comment by pik** [comment]
Yeah it should be doable. If anyone wants to do the glade for this that would be great (I promise to do the rest of the plugin).
